File: week11/20/pr4_Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        ans = ""
        if ans in strs: return ans
        
        temp = strs[0]
        while temp:
            ch = temp[0]
            for i in range(1,len(strs)):
                if not strs[i].startswith(ch):
                    return ans
                else : strs[i] = strs[i][1:]
            ans += ch
            temp = temp[1:]
        return ans

File: week11/20/pr4_Longest Common Prefix.c
char* longestCommonPrefix(char** strs, int strsSize) {
    int len = strlen(strs[0]);
    char* ans = (char*)malloc(200*sizeof(char));
    char* temp = strs[0];
    int idx = 0;
    for(int i=0;i<len;i++){
        idx = i;
       for(int j=0;j<strsSize;j++){
        if(i == strlen(strs[j])||strs[j][i]!=temp[i]){
            ans[idx] = '\0';
            return ans;
        }
       }
       ans[i] = temp[i];
    }
    return temp;
}